internal/server/receiver_server.go

```diff
@@ -71,3 +82,10 @@ func (s *ReceiverServer) ListenAndServe(stopCh <-chan struct{}) {
 		s.logger.Info("Receiver server stopped")
 	}
 }
+
+func receiverKeyFunc(r *http.Request) (string, error) {
+	id := url.PathEscape(strings.TrimLeft(r.RequestURI, "/hook/"))
+	val := strings.Join([]string{"receiver", id}, "/")
```

Comment:
We need to checksum the request body if it exists, as it is, we’ll block valid push webhooks for different commits/repos.

Reply:
We are pushing this issue forward as there will be complications with implementing rate limiting to the receiver. Instead we will solve it in a separate PR.